code/missionui/redalert.cpp

```diff
@@ -979,9 +985,15 @@ void red_alert_bash_wingman_status()
 						}
 					}
 
+					float max_hull;
+					if (pobjp->special_hitpoints)
+						max_hull = pobjp->ship_max_hull_strength;
+					else
+						max_hull = Ship_info[pobjp->ship_class].max_hull_strength;
```

Comment:
Since this code appears twice, perhaps a `get_max_hull()` function would be useful? Or are `pobjp` here and `shipp` above different types?

Reply:
They are different types.